### PR TITLE
api types: Sync RealmDataForUpdate with the recent update of the event doc

### PR DESCRIPTION
**Pull request by chrisbobbe**
In particular, zulip/zulip@864fb7c97 and some previous commits.

See discussion at
  https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60value.60.20in.20realm.20op.3A.20update/near/1282692.

### CODE REVIEW

**Comment by chrisbobbe** [comment]
Ah, and: because of that update in the API docs, we now know [what's happening with `push_notifications_enabled`](https://chat.zulip.org/#narrow/stream/378-api-design/topic/.60value.60.20in.20realm.20op.3A.20update/near/1274989). It doesn't get updated in an event.

So I've come back empty-handed for that bit of followup to #5090, which prompted all this work. Still, it's nice to have done it. I'll go and remove the TODO from my branch for #5090.

File: src/api/realmDataTypes.js [hunk 1]
```diff
@@ -181,64 +103,22 @@ export type RealmDataForUpdate = $ReadOnly<{
     $PropertyType<InitialDataRealm, 'realm_night_logo_url'>,
   notifications_stream_id:
     $PropertyType<InitialDataRealm, 'realm_notifications_stream_id'>,
-  password_auth_enabled:
-    $PropertyType<InitialDataRealm, 'realm_password_auth_enabled'>,
-  plan_type:
-    $PropertyType<InitialDataRealm, 'realm_plan_type'>,
```
Comment (gnprice) [inline src/api/realmDataTypes.js:187]: This one appears to still be in the doc.

**Review by gnprice — COMMENTED** [review]
Thanks for taking care of this! I looked through this comparing with those zulip/zulip commits, and with the current doc. Just one small comment below.

**Comment by chrisbobbe** [comment]
Eep, I forgot we hadn't finished this. Merged, with that fix.

**Pull request merged**